src/foobar/rest/views/purchase.py

```diff
@@ -1,30 +1,58 @@
 from rest_framework import viewsets, status
+from rest_framework.exceptions import NotFound
 from rest_framework.response import Response
 from authtoken.permissions import HasTokenScope
 
 from foobar import api
 from ..serializers.purchase import (
-    PurchaseRequestSerializer,
-    PurchaseSerializer
+    PurchaseSerializer,
+    PurchaseStatusSerializer,
+    PurchaseRequestSerializer
 )
 from wallet.exceptions import InsufficientFunds
 
 
 class PurchaseAPI(viewsets.ViewSet):
     permission_classes = (HasTokenScope('purchases'),)
 
+    def list(self, request):
+        purchases = api.list_purchases(request.query_params.get('account_id'))
```

Comment:
The query params should be passed through a serializer so that we can perform validation before inputing the values directly into to the internal API calls.